### ISSUE TITLE
Toggling aria-hidden on a parent element causes aria-hidden child element to be read

### ISSUE THREAD

**Issue by claydiffrient** [nvaccess/nvda#6053]
I've only seen this happen with NVDA 2016.2 and Firefox 47.  I was unable to reproduce the issue in the previous version of NVDA with an older Firefox (32).

I've produced a demo of the problem here (including workaround): http://jsbin.com/zihupokube/edit?html,css,js

In several modal libraries including (https://github.com/reactjs/react-modal/) the common practice is to apply `aria-hidden` to an "application" level container so that when a modal is open the content behind the modal is invisible to screenreaders.  When the modal is closed this is removed and the application content is again visible to screenreaders.

Unfortunately now when this occurs other content that is aria-hidden inside the application is now read by NVDA.  The workaround unfortunately is to force a DOM redraw which then makes everything work, however this shouldn't be necessary.

I would expect that NVDA respect the aria-hidden attribute on child components regardless of the aria-hidden status of the parent.

**Comment by Adriani90** [comment]
@claydiffrient are you still able to reproduce this issue? @instructure-gerrit your thoughts here would be very appreciated. I saw you did some work on this before. Thank you very much.
cc: @jcsteh

**Comment by jcsteh** [comment]
I can't reproduce the issue with the demo provided by the reporter. This doesn't surprise me; aria-hidden was completely re-implemented in Firefox last year and there are probably relevant NVDA fixes as well. Closing.
